File: src/main/java/com/yauhenl/poe/domain/Stash.java
package com.yauhenl.poe.domain;

import org.bson.Document;

import java.util.List;

public class Stash {

    public static Boolean isPublic(Document stash) {
        return stash.getBoolean("public");
    }

    @SuppressWarnings("unchecked")
    public static List<Document> getItems(Document stash) {
        return stash.get("items", List.class);
    }

    public static void setItems(Document stash, List<Document> items) {
        stash.put("items", items);
    }

    public static String getId(Document stash) {
        return stash.getString("id");
    }

    public static Boolean isVerified(Document item) {
        return item.getBoolean("verified");
    }

    public static String getAccountName(Document stash) {
        return stash.getString("accountName");
    }

    public static String getLastCharacterName(Document stash) {
        return stash.getString("lastCharacterName");
    }
}

File: src/main/kotlin/com/yauhenl/poe/domain/Stash.kt
package com.yauhenl.poe.domain

import org.bson.Document

object Stash {
    fun isPublic(stash: Document) = stash.getBoolean("public")

    @Suppress("UNCHECKED_CAST")
    fun getItems(stash: Document) = stash.get("items", List::class.java) as List<Document>

    fun setItems(stash: Document, items: List<Document>) = stash.put("items", items)

    fun getId(stash: Document) = stash.getString("id")

    fun isVerified(item: Document) = item.getBoolean("verified")

    fun getAccountName(stash: Document) = stash.getString("accountName")

    fun getLastCharacterName(stash: Document) = stash.getString("lastCharacterName")
}